blake3.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019, 2020 OCI Contributors
+// Copyright 2017 Docker, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package digest
+
+import (
+	"hash"
+
+	"github.com/zeebo/blake3"
```

Comment:
I'm not sure we want blake3 dependency in this repo (until blake3 becomes popular in the OCI ecosystem).

Instead, can we add functions for dynamically adding algorithms?

e.g.,
```go
package digest

func Register(algorithm string, siz int, newHash func ()hash.Hash)

```

Reply:
oh clever

Reply:
I can do that refactor.